fix: remove bulk instance migration feature due to server loading concerns

mas-who opened the pull request:
## Done

- From the discussion with lxd team, there are concerns about potentially overloading the server if bulk migration is performed on a large number of instances. For now, we will remove the feature and wait for the team to add background task scheduling capability so we can do bulk actions in the UI without issues.

Fixes [list issues/bugs if needed]

## QA

1. Run the LXD-UI:
    - On the demo server via the link posted by @webteam-app below. This is only available for PRs created by collaborators of the repo. Ask @mas-who or @edlerd for access.
    - With a local copy of this branch, [build and run as described in the docs](../CONTRIBUTING.md#setting-up-for-development).
2. Perform the following QA steps:
    - For the UI connected to a lxd cluster, check that the `Migrate` button is no longer visible when selecting multiple instances on the instance list page.

webteam-app commented:
### [<img src='https://assets.ubuntu.com/v1/6baef514-ubuntu-circle-of-friends-large.svg' height=32 width=32> Demo</img>](https://lxd-ui-869.demos.haus)
### [<img src='https://assets.ubuntu.com/v1/e512b0e2-jenkins.svg' height=32 width=32> Jenkins </img>](https://jenkins.canonical.com/webteam/job/start-demo/13019)
### [<img src='https://assets.ubuntu.com/v1/7144ec6d-logo-jaas-icon.svg' height=32 width=32> demos.haus </img>](https://demos.haus)

Kxiru reviewed (commented):
Big oof. Looks good to me though. After all, it's not affecting anything else that is there. I'll allow David to give a final review.

edlerd reviewed (commented):
I am not a fan. If we remove it, we should also clean up the `isBulkMigration` from the `MigrateInstanceModal` and `migrateInstanceBulk` from the `api/instances.tsx` file.

mas-who commented:
> I am not a fan. If we remove it, we should also clean up the `isBulkMigration` from the `MigrateInstanceModal` and `migrateInstanceBulk` from the `api/instances.tsx` file.

Yeah I'm also not a huge fan of this change, but I think overloading the server is a valid point. If you don't want to merge this one, maybe when I get back we can look into processing batches of requests instead. In the interim, I've done some more cleaning up, removed all bulk migration related logic and also adjusted the modal component to process a single instance only.
